src/multi.c

```diff
@@ -74,7 +74,7 @@ void queueMultiCommand(client *c, uint64_t cmd_flags) {
     }
     if (c->mstate.count == c->mstate.alloc_count) {
         c->mstate.alloc_count = c->mstate.alloc_count < INT_MAX/2 ? c->mstate.alloc_count*2 : INT_MAX;
-        c->mstate.commands = zrealloc(c->mstate.commands, sizeof(multiCmd)*(c->mstate.alloc_count*2));
+        c->mstate.commands = zrealloc(c->mstate.commands, sizeof(multiCmd)*(c->mstate.alloc_count));
```

Comment:
It looks like the same overflow will occur here (32*INT_MAX), which you didn't convert to size_t first.

Reply:
```
#include <stdio.h>

typedef struct {
	int a;
	int b;
	int c;
	int d;
} test;

int main(void) {
	// your code goes here
	printf("%zu", sizeof(test) * 0x7fffffff);

	return 0;
}
```

On both 32/64 bit it works fine, verified locally. `size_t` type is taking higher precedence.

Also, we have similar code snippet in Redis already: https://github.com/redis/redis/blob/unstable/src/networking.c#L2357-L2358